block-layer/graphsync/known_extensions.md

```diff
@@ -0,0 +1,63 @@
+# Graphsync: Known Extensions
+
+### Excluded CIDs
```

Comment:
* Does this exclude entire sub-trees (i.e., stop traversal) or just the node in question? IMO, this is the most _useful_ mode of operation.
* Can I _update_ the request with additional exclusions? If not, this feature feels like it belongs inside selectors themselves. @warpfork?

Reply:
So, to be clear, excluded cids -- and maybe it needs another name -- do not affect the execution of the query itself. It does not mean "stop at". Instead, it means "when you pass over this block during traversal, there's no need to send it back to me, cause I already have it". It's purely for network efficiency -- the main use case is I execute a graphsync query that had a network interruption half way through and I want to redo the request without sending the first half of the data.

My belief is this kind of data-efficiency exclusion is pretty tough to express in a selector query  -- there is an add-on coming to selectors that specifically includes the kind of "stopAt" in a recursive selector that you're talking about.

Reply:
Ah, and we can't bundle it into a _selector_ because we need to send back the entire selector trace, even if a block doesn't match. Makes sense.